app/models/annotation_text.rb

```diff
@@ -3,15 +3,58 @@ class AnnotationText < ApplicationRecord
   belongs_to :creator, class_name: 'User', foreign_key: :creator_id
   belongs_to :last_editor, class_name: 'User', foreign_key: :last_editor_id, optional: true
 
+  after_update :update_mark_deductions
+  before_update :check_if_released
+  before_destroy :check_if_released
+
   # An AnnotationText has many Annotations that are destroyed when an
   # AnnotationText is destroyed.
   has_many :annotations, dependent: :destroy
 
   belongs_to :annotation_category, optional: true, counter_cache: true
   validates_associated :annotation_category, on: :create
 
+  validates_numericality_of :deduction,
+                            if: :should_have_deduction?,
+                            greater_than_or_equal_to: 0,
+                            less_than_or_equal_to: ->(t) { t.annotation_category.flexible_criterion.max_mark }
+
+  validates_absence_of :deduction, unless: :should_have_deduction?
+
+  def should_have_deduction?
+    !self&.annotation_category&.flexible_criterion_id.nil?
+  end
+
   def escape_content
     content.gsub('\\', '\\\\\\') # Replaces '\\' with '\\\\'
            .gsub(/\r?\n/, '\\n')
   end
+
+  def check_if_released
+    # Cannot update if any results have been released with the annotation and the deduction is non nil
+    return if self.annotations.joins(:result).where('results.released_to_students' => true).empty? ||
+        self.deduction.nil?
```

Comment:
We will keep this as limited as it is for now, because the development of the platform is hopefully moving in the direction of this functionality holding true for all annotations.